src/Aspire.Hosting/Publishing/ManifestPublisher.cs

```diff
@@ -136,6 +136,22 @@ private static void WriteServiceDiscoveryEnvironmentVariables(IResource resource
             }
         }
     }
+    internal static void WriteDependencies(IResource resource, ManifestPublishingContext context)
```

Comment:
Need to consider how to reuse this logic for custom publishing callbacks outside this assembly.